### PR TITLE
MNMG - Initial Feature Branch

### LINKED ISSUE #336: MNMG - Refactor Control Plane to use Redux/Node/Typescript

### LINKED ISSUE #338: MNMG - Backend C++ updates to use new Control Plane

### PR DESCRIPTION
**Pull request by mdemoret-nv**
This PR will merge the initial changes required for MNMG into a feature branch to help track completion on the large list of issues required for MNMG.

This PR contains:

- The initial NodeJS based control plane
- Refactoring of resources
- Creation of C++ components to match the Control Plane State
- Startup/shutdown of resources using `AsyncService` 

Closes #336 
Closes #338

### CODE REVIEW

File: Dockerfile copy [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# syntax=docker/dockerfile:1.3
```
Comment (dagardner-nv) [inline Dockerfile copy:1]: Was this committed by mistake?

File: vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (dagardner-nv) [inline vcpkg.json:1]: Do we still support vcpkg?

**Review by dagardner-nv — APPROVED** [review]

File: vcpkg.json (hunk 2 above)
Comment (mdemoret-nv) [reply vcpkg.json:1]: No we dont but sometimes I add it back in to get debug builds of boost which helps with debugging fibers issues. Will remove

File: Dockerfile copy (hunk 1 above)
Comment (mdemoret-nv) [reply Dockerfile copy:1]: Yes, removed

**Comment by mdemoret-nv** [comment]
Merging without CI since this is a feature branch

**Pull request merged**